pkg/common/resources/resources_test.go

```diff
@@ -17,39 +17,858 @@ limitations under the License.
+    }
+    base = NewResourceFromMap(map[string]Quantity{"first": 1})
+    compare = NewResourceFromMap(map[string]Quantity{"first": 1, "second": 1})
+    if Equals(base, compare) {
+        t.Errorf("compared resources are equal, should not be: %v / %v", base, compare)
+    }
+}
+
+func TestIsZero(t *testing.T) {
+    // simple cases (nil checks)
+    if !IsZero(nil) {
+        t.Errorf("nil resource should be zero")
+    }
+    base := NewResourceFromMap(map[string]Quantity{})
+    if !IsZero(base) {
+        t.Errorf("no resource entries should be zero")
+    }
+
+    // set resource values
+    base = NewResourceFromMap(map[string]Quantity{"zero": 0})
+    if !IsZero(base) {
+        t.Errorf("only zero resources should be zero")
+    }
+    base = NewResourceFromMap(map[string]Quantity{"first": 10})
+    if IsZero(base) {
+        t.Errorf("set resource should be non zero")
+    }
+}
+
+func TestStrictlyGreaterThanZero (t *testing.T) {
+    // simple case (nil checks)
+    if StrictlyGreaterThanZero(nil) {
+        t.Errorf("nil resource should not be greater than zero")
+    }
+    base := NewResourceFromMap(map[string]Quantity{})
+    if StrictlyGreaterThanZero(base) {
+        t.Errorf("no resource entries should not be greater than zero")
+    }
+
+    // set resource values
+    base = NewResourceFromMap(map[string]Quantity{"zero": 0})
+    if StrictlyGreaterThanZero(base) {
+        t.Errorf("only zero resources should not be greater than zero")
+    }
+    base = NewResourceFromMap(map[string]Quantity{"first": 10})
+    if !StrictlyGreaterThanZero(base) {
+        t.Errorf("set resource should be greater than zero")
+    }
+    base = NewResourceFromMap(map[string]Quantity{"first": -1})
+    if StrictlyGreaterThanZero(base) {
+        t.Errorf("set negative resource should not be greater than zero")
+    }
+}
+
+func TestStrictlyGreaterThan (t *testing.T) {
+    // simple case (nil checks)
+    if StrictlyGreaterThan(nil, nil) {
+        t.Errorf("nil resource should not be greater than other nil")
+    }
+    // zero or empty resource values
+    larger := NewResourceFromMap(map[string]Quantity{})
+    smaller := NewResourceFromMap(map[string]Quantity{})
+    if StrictlyGreaterThan(larger, smaller) {
+        t.Errorf("empty resource entries should not be greater than empty")
+    }
+    larger = NewResourceFromMap(map[string]Quantity{"zero": 0})
+    if StrictlyGreaterThan(larger, smaller) {
+        t.Errorf("only zero resources should not be greater than empty")
+    }
+    smaller = NewResourceFromMap(map[string]Quantity{"other": 0})
+    if StrictlyGreaterThan(larger, smaller) {
+        t.Errorf("only zero resource (both objects) should not be greater than zero")
+    }
+
+    // negative resource values
+    larger = NewResourceFromMap(map[string]Quantity{})
+    smaller = NewResourceFromMap(map[string]Quantity{"first": -1})
+    if !StrictlyGreaterThan(larger, smaller) {
```

Comment:
We cannot do that as it would go against the fact that resources are defined as sparse objects. Resources by definition are considered sparse, which means any resource value not defined is considered a zero (0). If we step away from resources being sparse we would also need to change all other comparisons like `Equal` and `GreaterThan` etc and make sure that every possible resource type is defined in every single resource object.

Reply:
Oh, this makes sense.
So here `larger` is same to  `map[string]Quantity{"first": 0}`
when it compares to `smaller`, which means `larger > smaller`.
I was confused when a resource value becomes to be negative, I am good with this now.
Thanks for adding these test cases!